drivers/i2c/i2c_ll_stm32.c

```diff
@@ -58,23 +58,29 @@ static int i2c_stm32_transfer(struct device *dev, struct i2c_msg *msg,
 	I2C_TypeDef *i2c = cfg->i2c;
 	int ret = 0;
 
-	LL_I2C_Enable(i2c);
-
+	/* Check for validity of all messages, to prevent having to abort
+	 * in the middle of a transfer
+	 */
```

Comment:
Yes, this could be in a helper shred by all i2c controllers instead, with some parameters like the max messgae length.